paprika/restraints/utils.py

```diff
@@ -158,3 +158,211 @@ def get_bias_potential_type(restraint, phase, window_number):
+        list of guest-specific :class:`paprika.restraints.DAT_restraint` restraints.
+
+    Examples
+    --------
+        Extract guest restraints from a list of restraints and compute the work for reference state
+
+        >>> guest_restraints = extract_guest_restraints(structure, restraints, "BEN")
+        >>>
+        >>> free_energy = analysis.fe_calc()
+        >>> free_energy.compute_ref_state_work(guest_restraints)
+        >>> print(free_energy["ref_state_work"])
+
+    """
+    guest_resname = guest_resname.upper()
+
+    DM1 = f"{dummy_prefix}1"
+    DM2 = f"{dummy_prefix}2"
+    DM3 = f"{dummy_prefix}3"
+
+    r = None
+    theta = None
+    phi = None
+    alpha = None
+    beta = None
+    gamma = None
+
+    for restraint in restraints:
+
+        mask2_residue_name = structure[restraint.mask2].residues[0].name
+
+        # Distance
+        if (
+            DM1 in restraint.mask1
+            and guest_resname in mask2_residue_name
+            and not restraint.mask3
+            and not restraint.mask4
+        ):
+            r = restraint
+
+        # Angle
+        if restraint.mask3 and not restraint.mask4:
+            mask3_residue_name = structure[restraint.mask3].residues[0].name
+
+            if (
+                DM2 in restraint.mask1
+                and DM1 in restraint.mask2
+                and guest_resname in mask3_residue_name
+            ):
+                theta = restraint
+
+            if (
+                DM1 in restraint.mask1
+                and guest_resname in mask2_residue_name
+                and guest_resname in mask3_residue_name
+            ):
+                beta = restraint
+
+        # Dihedral
+        if restraint.mask4:
+            mask3_residue_name = structure[restraint.mask3].residues[0].name
+            mask4_residue_name = structure[restraint.mask4].residues[0].name
+
+            if (
+                DM3 in restraint.mask1
+                and DM2 in restraint.mask2
+                and DM1 in restraint.mask3
+                and guest_resname in mask4_residue_name
+            ):
+                phi = restraint
+
+            if (
+                DM2 in restraint.mask1
+                and DM1 in restraint.mask2
+                and guest_resname in mask3_residue_name
+                and guest_resname in mask4_residue_name
+            ):
+                alpha = restraint
+
+            if (
+                DM1 in restraint.mask1
+                and guest_resname in mask2_residue_name
+                and guest_resname in mask3_residue_name
+                and guest_resname in mask4_residue_name
+            ):
+                gamma = restraint
+
+    guest_restraints = [r, theta, phi, alpha, beta, gamma]
+
+    return guest_restraints
+
+
+def restraints_from_ascii(filename):
+    """
+    Utility function to read in restraints from a simple ASCII file. This is
+    useful when parsing restraints definition from VMD. Since you can mouse-click
```

Comment:
Suggested rephrasing:

"This is useful when parsing restraints definition from VMD" → "This is useful for using VMD to define restraints."

Reply:
fixed